### ISSUE TITLE
missing cpu_set_t on Alpine linux (musl libc)

### ISSUE THREAD

**Issue by dimpase** [wbhart/flint2#1054]
From https://trac.sagemath.org/ticket/33083 
```
In file included from /sage/local/var/tmp/sage/build/flint-2.7.1/src/thread_support.h:16,
                 from /sage/local/var/tmp/sage/build/flint-2.7.1/src/nmod_mat.h:35,
                 from /sage/local/var/tmp/sage/build/flint-2.7.1/src/nmod_poly.h:34,
                 from /sage/local/var/tmp/sage/build/flint-2.7.1/src/fmpz_poly.h:35,
                 from /sage/local/var/tmp/sage/build/flint-2.7.1/src/aprcl.h:18,
                 from config_gauss.c:12:
/sage/local/var/tmp/sage/build/flint-2.7.1/src/thread_pool.h:55:5: error: unknown type name 'cpu_set_t'
   55 |     cpu_set_t original_affinity;
      |     ^~~~~~~~~
```
Probably as a workaround one can put `-D_GNU_SOURCE` into `CFLAGS`.

**Comment by dimpase** [comment]
Indeed, this is a [GNU extension](https://www.gnu.org/software/libc/manual/html_node/CPU-Affinity.html), no wonder.
```
Data Type: cpu_set_t
This data set is a bitset where each bit represents a CPU. How the system’s CPUs are mapped to bits in the bitset is system dependent. The data type has a fixed size; in the unlikely case that the number of bits are not sufficient to describe the CPUs of the system a different interface has to be used.

This type is a GNU extension and is defined in sched.h.
```

**Comment by dimpase** [comment]
This is to be fixed by #988
